go/app/main.go

```diff
@@ -20,19 +25,84 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type Item struct {
+	Name         string `json:"name"`
+	Category     string `json:"category"`
+	Img_filename string `json:"img_filename"`
+}
+
+type ItemWrapper struct {
+	Items []Item `json:"items"`
+}
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
+func hashString(s string) string {
+	h := sha256.New()
+	h.Write([]byte(s))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
 func addItem(c echo.Context) error {
 	// Get form data
 	name := c.FormValue("name")
-	c.Logger().Infof("Receive item: %s", name)
+	category := c.FormValue("category")
+	imagePath := c.FormValue("image")
+
+	// Hash image
+	imgFileName := filepath.Base(imagePath)
+	img := strings.TrimSuffix(imgFileName, filepath.Ext(imgFileName))
+	hash := hashString(img)
```

Comment:
[nits] 単にhashだけだと何のhashかわからないかもしれないので、変数名を変えるのはどうでしょうか？

```suggestion
	hashImageName := hashString(img)
```